jdbc/src/java/org/apache/hive/jdbc/HttpRequestInterceptorBase.java

```diff
@@ -77,6 +88,14 @@ public void process(HttpRequest httpRequest, HttpContext httpContext)
       if (isCookieEnabled) {
         httpContext.setAttribute(Utils.HIVE_SERVER2_RETRY_KEY, Utils.HIVE_SERVER2_CONST_FALSE);
       }
+
+      if (requestTrackingEnabled) {
+        String trackHeader = getNewTrackHeader();
+        LOG.info("{}:{}", Constants.HTTP_HEADER_REQUEST_TRACK, trackHeader);
+        additionalHeaders.put(Constants.HTTP_HEADER_REQUEST_TRACK, trackHeader);
+        httpContext.setAttribute(Constants.HTTP_HEADER_REQUEST_TRACK, trackHeader);
+        httpContext.setAttribute(trackHeader + Constants.TIME_POSTFIX_REQUEST_TRACK, System.currentTimeMillis());
```

Comment:
We are using ``System.currentTimeMillis()`` to calculate the duration. I am not very sure, if using Time.monotonicNow() Will be better.
Just by the doc:
https://github.com/apache/hadoop/blob/b1f418f8027c2d2a1a1c17ec3a8d23ceaaf8d89c/hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/util/Time.java#L49-L69

Reply:
yeah, I tend to use System.currentTimeMillis(), hadoop util looks better clearly according to javadoc